test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -1038,10 +1038,10 @@ def addSuccess(self, test):
     def _cleanup(self):
         self.logger.info('Cleaning up temporary files')
         du = DshUtils()
-        hosts = self.param_dict['moms']
+        hosts = list(self.param_dict['moms'])
```

Comment:
Why are u converting `hosts` from `set` to `list`? we need it as set as we want to keep unique keys in it.

Reply:
Because if we use hosts = self.param_dict['moms'] and in later step we do hosts.update(server) then not only hosts updated with server but also origin self.param_dict['moms']. that is the set behavior. 

We can get unique value by using if server not in hosts:

Reply:
for that you just need to create copy of self.param_dict['moms'] using `hosts = set(self.param_dict['moms'])`